File: back-end/src/main/java/es/ull/project/domain/valueobject/FileName.java
package es.ull.project.domain.valueobject;

import java.util.Objects;

/**
 * @brief Value object representing a file name.
 */
public class FileName {

    private static final String ERROR_FILE_NAME_NULL_OR_EMPTY = "File Name cannot be null or empty";
    private final String name;

    public FileName(String name) {
        this.validateName(name);
        this.name = name;
    }

    private void validateName(String name) {
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException(ERROR_FILE_NAME_NULL_OR_EMPTY);
        }
    }

    public String getName() {
        return this.name;
    }

    public FileName setName(String name) {
        return new FileName(name);
    }

    @Override
    public boolean equals(Object otherObject) {
        if (this == otherObject) {
            return true;
        }
        if (!(otherObject instanceof FileName)) {
            return false;
        }
        final FileName fileName = (FileName) otherObject;
        return Objects.equals(name, fileName.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name);
    }

    @Override
    public String toString() {
        return name;
    }
}

File: front-end/src/domain/value-object/file-name.ts
export class FileName {
    private static readonly ERROR_NAME_NOT_DEFINED = 'File Name cannot be null or empty';
    private readonly name: string;

    constructor(name: string) {
        this.validateName(name);
        this.name = name;
    }

    getName(): string {
        return this.name;
    }

    setName(name: string): FileName {
        return new FileName(name);
    }


    validateName(name: string): void {
        if (!name || name.trim() === '') {
            throw new Error(FileName.ERROR_NAME_NOT_DEFINED);
        }
    }

    equals(other: FileName): boolean {
        return other instanceof FileName && this.name === other.name;
    }

    toString(): string {
        return `FileName={${this.name}}`;
    }
}
